include/boost/json/parser.hpp

```diff
@@ -156,127 +157,137 @@ class parser
-        before allocating with the memory
-        resource specified on construction.
-
-        @par Example
-
-        The following code constructs a parser which
-        uses the default memory resource and a local
-        buffer for temporary storage, and allows
-        trailing commas to appear in the JSON:
-
-        @code
-
-        // this buffer will be used for temporary storage
-        unsigned char temp[ 4096 ];
-
-        // default constructed parse options allow strict JSON
-        parse_options opt;
-
-        // enable the trailing commas extension
-        opt.allow_trailing_commas = true;
-
-        // construct the parser
-        parser p(
-            storage_ptr(),  // use the default memory resource
-            opt,
-            temp, sizeof(temp) );
-
-        // to begin parsing, reset must becalled
-        p.reset();
-
-        @endcode
+        This constructs a new parser which first uses the
+        caller-owned storage pointed to by `temp_buffer`
+        for temporary storage, falling back to the memory
+        resource `sp` if needed. The parser will use the
+        specified parsing options.
+    \n
+        The parsed value will use the default memory resource
+        for storage. To use a different resource, call
+        @ref reset after construction.
 
         @par Complexity
         Constant.
 
         @par Exception Safety
         No-throw guarantee.
 
-        @param sp A pointer to the @ref memory_resource
-        for the implementation to use to acquire
-        temporary storage.
-
-        @param opt The options for the parser. If this
-        parameter is omitted, a default constructed
-        parse options is used, which allows only strict
-        JSON and an implementation defined maximum depth.
-
-        @param temp_buffer A pointer to valid memory
-        which the implementation will use first to
-        acquire temporary storage, or `nullptr` for
-        the implementation to go directoy to the
-        memory resource. If this parameter is left out
-        the behavior is the same as if it were null.
-
-        @param temp_size The size of the memory pointed
-        to by `temp_buffer`. This parameter is ignored
-        if `temp_buffer` is null.
+        @param sp The memory resource to use for temporary storage
+        when `buffer` is exhausted.
+
+        @param opt The parsing options to use.
+
+        @param buffer A pointer to valid memory of at least
+        `size` bytes for the parser to use for temporary storage.
+        Ownership is not transferred, the caller is responsible
+        for ensuring the lifetime of the memory pointed to by
+        `buffer` extends until the parser is destroyed.
     */
     BOOST_JSON_DECL
     parser(
         storage_ptr sp,
         parse_options const& opt,
-        unsigned char* temp_buffer,
-        std::size_t temp_size) noexcept;
+        unsigned char* buffer,
+        std::size_t size) noexcept;
```

Comment:
Have you considered using a string_view here... or is it just a matter of making calls using `std::byte*` less clunky?

Reply:
`string_view` is non-mutable

Reply:
Oh, my bad, I made the wrong assumption here... I guess that a span-like type could be useful here then, but if you don't have any such construct in this project, it doesn't pay off to bring it from another dependency.